packages/salesforcedx-vscode-core/src/providers/configAggregatorProvider.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { ConfigAggregator, SfdxConfigAggregator } from '@salesforce/core';
+import { WorkspaceContext } from '../context/workspaceContext';
+
+type ConfigAggregatorOptions = {
+  /*
+   *  The SfdxConfigAggregator is used only to get configuration
+   *  values that correspond with old/deprecated config keys.
+   *  Currently, the key used for the custom templates
+   *  directory is the only usage, since it is documented for use
+   *  here: https://developer.salesforce.com/tools/vscode/en/user-guide/byotemplate#set-default-template-location
+   */
+  sfdx?: boolean;
+};
+
+/*
+ * The ConfigAggregatorProvider class is used to instantiate
+ * ConfigAggregator and SfdxConfigAggregator in the VSCE context.
+ * In order to accurately retrieve both project and global variable values,
+ * It needs to be instantiated while process.cwd() returns the root project
+ * workspace path. ConfigAggregatorProvider handles this and switches the
+ * cwd back so that it is the same when the function finishes.
+ */
+export class ConfigAggregatorProvider {
+  protected configAggregators: Map<string, ConfigAggregator>;
+  protected sfdxConfigAggregators: Map<string, ConfigAggregator>;
+  protected globalConfigAggregator: ConfigAggregator | undefined = undefined;
```

Comment:
Any specific reason for defining undefined like this? Why can't we just go with `ConfigAggregator | undefined`

Reply:
updated!